src/mono/mono/mini/aot-compiler.c

```diff
@@ -13142,77 +13211,166 @@ add_profile_instances (MonoAotCompile *acfg, ProfileData *data)
 		MonoMethod *m = mdata->method;
-		MonoGenericContext *ctx;
+		count += add_single_profile_method (acfg, m);
+	}
+
+	printf ("Added %d methods from profile.\n", count);
+}
+
+typedef enum {
+	FIND_METHOD_TYPE_ENTRY_START,
+	FIND_METHOD_TYPE_ENTRY_END,
+} MibcGroupMethodEntryState;
+
+//---------------------------------------------------------------------------------------
+//
+// add_mibc_group_method_methods iterates over a mibcGroupMethod MonoMethod to obtain
+// each method/type entry in that group. Each entry begins with LDTOKEN opcode, followed
+// by a series of instructions including another LDTOKEN opcode but excluding a POP opcode
+// ending with a POP opcode to denote the end of the entry. To iterate over entries,
+// a MibcGroupMethodEntryState state is tracked, identifying whether we are looking for
+// an entry starting LDTOKEN opcode or entry ending POP opcode. Each LDTOKEN opcode's
+// argument denotes a methodref or methodspec. Before ultimately adding the entry's method
+// for AOT compilation, checks for the method's class and image are used to skip methods
+// whose class and images cannot be resolved.
+//
+// Sample mibcGroupMethod format
+//
+// Method 'Assemblies_HelloWorld;_1' (#11b) (0x06000001)
+// {
+//   IL_0000:  ldtoken    0x0A0002B5 // Method/Type Entry token
+//   IL_0005:  pop
+// }
+//
+// Arguments:
+// 	* acfg - MonoAotCompiler instance
+//	* mibcGroupMethod - MonoMethod representing the group of assemblies that
+//				describes each Method/Type entry with the .mibc file
+//	* image - MonoImage representing the .mibc file
+//	* mibcModuleClass - MonoClass containing the AssemblyDictionary
+//	* context - MonoGenericContext of the AssemblyDictionary MonoMethod
+//
+// Return Value:
+//	int pertaining to the number of methods added within the mibcGroupMethod
+//
+
+static int
+add_mibc_group_method_methods (MonoAotCompile *acfg, MonoMethod *mibcGroupMethod, MonoImage *image, MonoClass *mibcModuleClass, MonoGenericContext *context)
+{
+	ERROR_DECL (error);
 
-		if (!m)
+	MonoMethodHeader *mibcGroupMethodHeader = mono_method_get_header_internal (mibcGroupMethod, error);
+	mono_error_assert_ok (error);
+
+	int count = 0;
+	MibcGroupMethodEntryState state = FIND_METHOD_TYPE_ENTRY_START;
+	uint8_t *cur = (uint8_t*)mibcGroupMethodHeader->code;
+	uint8_t *end = (uint8_t*)mibcGroupMethodHeader->code + mibcGroupMethodHeader->code_size;
+	while (cur < end) {
+		MonoOpcodeEnum il_op;
+		const unsigned char *opcodeIp = (unsigned char*)cur;
+		const unsigned char *opcodeEnd = (unsigned char*)end;
+		cur += mono_opcode_value_and_size (&opcodeIp, opcodeEnd, &il_op);
+
+		if (state == FIND_METHOD_TYPE_ENTRY_END) {
+			if (il_op == MONO_CEE_POP)
+				state = FIND_METHOD_TYPE_ENTRY_START;
 			continue;
-		if (!m->is_inflated)
+		}
+		state = FIND_METHOD_TYPE_ENTRY_END;
+
+		g_assert (opcodeIp + 4 < opcodeEnd);
+		guint32 mibcGroupMethodEntryToken = *(guint32 *)(opcodeIp + 1);
```

Comment:
```suggestion
		guint32 mibcGroupMethodEntryToken = read32 ((opcodeIp + 1));
```

Need to be careful about endianness and unaligned access.